Library/Homebrew/cmd/--env.rb

```diff
@@ -3,19 +3,37 @@
 
 require "extend/ENV"
 require "build_environment"
+require "utils/shell"
 
 module Homebrew
+  def explicit_shell
+    shell = ARGV.value("shell")
+    return :noshell if shell == "none"
```

Comment:
More lines, but (in my opinion) easier to read would be to make this and the following line:

```ruby
if shell == "none"
  :noshell
elsif shell
  Utils::Shell.path_to_shell(shell)
end
```